app/models/goods_nomenclatures/tree_node.rb

```diff
@@ -8,6 +17,39 @@ class << self
       def refresh!(concurrently: true)
         db.refresh_view(:goods_nomenclature_tree_nodes, concurrently:)
       end
+
+      def previous_sibling(origin_position, origin_depth)
+        siblings_position = Sequel.qualify(:siblings, :position)
+        siblings_depth    = Sequel.qualify(:siblings, :depth)
+        siblings_table    = Sequel.as(table_name, :siblings)
+
+        from(siblings_table)
+          .select { Sequel.as(max(siblings_position), :previous_sibling) }
+          .where do |_query|
+            (siblings_depth =~ origin_depth) &
+              (siblings_position < origin_position) &
+              (siblings_position >= TreeNode.start_of_chapter(origin_position)) &
+              validity_dates_filter(:siblings)
+          end
+      end
+
+      def start_of_chapter(position_column)
+        (position_column / 10_000_000_000) * 10_000_000_000
```

Comment:
Would be good to give this number a name

Reply:
Yeh, good should mean to circle back to this and forgot